src/error_estimation/adjoint_refinement_estimator.C

```diff
@@ -507,11 +507,9 @@ void AdjointRefinementEstimator::estimate_error (const System & _system,
               // We will have to manually do the dot products.
               Number local_contribution = 0.;
 
-              for (std::size_t j=0; j != dof_indices.size(); j++)
-                {
-                  // The contribution to the error indicator for this element from the current QoI
-                  local_contribution += (*localized_projected_residual)(dof_indices[j]) * (*localized_adjoint_solution)(dof_indices[j]);
-                }
+              // Sum the contribution to the error indicator for each element from the current QoI
+              for (const auto & dof : dof_indices)
+                local_contribution += Utility::pow<2>((*localized_projected_residual)(dof));
```

Comment:
No pow here.  Those are two different vectors being combined.

Reply:
Although, and this is damning, this looks like it might be *triggering* but not *causing* the GRINS error.  I'll have to try this change out on master myself some time and see if I can replicate.

Reply:
Oh crap, my eyes must have been going crossed when I changed that line. Will fix.

Reply:
GRINS seems happier after this fix. I'm glad we have the extra coverage!

Reply:
No kidding.  I can't even quantify the extra coverage here, though - libMesh "make check" uses this code in two examples!  I guess the trouble is it's an error indicator and libMesh examples aren't have good enough regression tests to tell us if the refinement patterns are hosed.

Reply:
Agreed, our reliance on dependent libraries is pretty embarrassing, but who really has the time/desire to make the examples into a real regression test suite with gold files, etc.?